pkg/machines/components/networks/createNetworkDialog.jsx

```diff
@@ -0,0 +1,608 @@
+        </React.Fragment>
+    );
+};
+
+const NetworkDeviceRow = ({ onValueChanged, dialogValues, vms, nodeDevices, networks, validationFailed }) => {
+    const validationState = validationFailed.physicalDevice ? 'error' : undefined;
+    const devices = getNetworkPhysicalDevices(vms, nodeDevices, networks);
+
+    // TODO Implement "Any physical device" option later
+    return (
+        <React.Fragment>
+            <label className='control-label'>
+                {_("Device")}
+            </label>
+            <FormGroup validationState={validationState} controlId='physicalDevice'>
+                <Select.Select id='create-network-physical-device'
+                               enabled={dialogValues.physicalDevice !== undefined}
+                               initial={dialogValues.physicalDevice !== undefined || "No Physical Device Available"}
+                               onChange={value => onValueChanged('physicalDevice', value)}>
+                    { devices.length > 0 ? devices.map(dev => {
+                        return (
+                            <Select.SelectEntry data={dev} key={dev}>
+                                {dev}
+                            </Select.SelectEntry>
+                        );
+                    }) : [<Select.SelectEntry data='no-resource' key='no-resource'>
+                        {_("No Physical Device Available")}
+                    </Select.SelectEntry> ]
+                    }
+                </Select.Select>
+                { validationState == 'error' &&
+                <HelpBlock>
+                    <p className='text-danger'>{validationFailed.physicalDevice}</p>
+                </HelpBlock> }
+            </FormGroup>
+        </React.Fragment>
+    );
+};
+
+const IpRow = ({ onValueChanged, dialogValues }) => {
+    return (
+        <React.Fragment>
+            <label className='control-label'>
+                {_("IP Configuration")}
+            </label>
+            <Select.Select id='create-network-forward-mode'
+                           initial={dialogValues.ip}
+                           onChange={value => onValueChanged('ip', value)}>
+                { (dialogValues.forwardMode === "none") &&
+                <Select.SelectEntry data='None' key='None'>
+                    {_("None")}
+                </Select.SelectEntry>}
+                <Select.SelectEntry data='IPv4 only' key='IPv4 only'>
+                    {_("IPv4 only")}
+                </Select.SelectEntry>
+                <Select.SelectEntry data='IPv6 only' key='IPv6 only'>
+                    {_("IPv6 only")}
+                </Select.SelectEntry>
+                <Select.SelectEntry data='IPv4 and IPv6' key='IPv4 and IPv6'>
+                    {_("IPv4 and IPv6")}
+                </Select.SelectEntry>
+            </Select.Select>
+        </React.Fragment>
+    );
+};
+
+const DhcpRow = ({ ipVersion, rangeStart, rangeEnd, expanded, onValueChanged, validationFailed }) => {
```

Comment:
Here you allow the dhcpv4 settings.
However libvirt allows you to configure dhcpv6 as well.
Virt-manager has clear the difference between the two.
Do you think this deserves some clarification?

Reply:
I don't know what clarification do you mean. This component is called from 2 places (IPv4Row and IPv6Row). I just use same component for that to not double code. 
Or do you mean make some clarification in the UI? In that case, I could change the text of related checkbox "Set DHCP Range" -> "Set IPvX DHCP Range".

Reply:
Sorry, didn't notice you reuse it.